spec/spec.md

```diff
@@ -0,0 +1,109 @@
+
+The spec will be worked on in the main branch and include the `-alpha` suffix until it reached a stable version. Once the first stable version of the spec is reached, it will be under the release-x.y branch and tagged with vX.Y.0, such as the runtime release process.
+
+This allows SDK developers to claim compliance towards a fixed spec and runtime version instead of a moving target.
+
+### SDK Versioning
+
+An SDK MUST start as one repository in the [dapr-sandbox organization](https://github.com/dapr-sandbox) at version 0.1, following the process to onboard a new Dapr sandbox project. Before an SDK can graduate to the [dapr organization](https://github.com/dapr), it MUST comply to a stable version of this spec, in addition to the graduation criteria for Dapr sandbox projects.
+
+## Client Object
+
+The SDK MUST offer a way to instantiate a client object without requiring any parameter, using the `DAPR_HTTP_PORT` or `DAPR_GRPC_PORT` environment variables injected by the Dapr sidecar injector or Dapr CLI - defaulting to `3500` or `50001`, correspondingly, if environment variable is not set.
+
+The client object MUST connect to the Dapr sidecar based on the `DAPR_SIDECAR_IP` environment variable, defaulting to `127.0.0.1` if not set.
+
+The client object MUST abstract the protocol used to communicate to the Dapr sidecar and not offer gRPC or HTTP specific input or output.
+
+The client object MIGHT offer both gRPC or HTTP implementations.
+
+The client object SHOULD NOT implement any Actor functionality since that is handled as a separate construct.
+
+The client object MUST implement all required methods for all [core building blocks](#core-building-blocks).
+
+The client object MIGHT provide overloaded methods with reduced number of parameters for simpler method invocations.
+
+The client object SHOULD NOT implement any of the methods for the [preview APIs](#alpha-apis)
+
+## Object Serialization
+
+The SDK MUST serialize domain objects to JSON by default.
+
+The SDK MIGHT accept a serializer implementation via a provided interface.
+
+The SDK MIGHT accept a separate state serializer implementation via a provided interface to differentiate serialization of transient domain objects vs persisted domain objects.
+
+## Exception Handling
+
+Behavior and structure for exceptions is not defined yet.
+
+## Core Building Blocks
+
+* [State Store](core/state-store.md)
+* [PubSub](core/pubsub.md)
+* [Input Binding](core/input-binding.md)
+* [Output Binding](core/output-binding.md)
+* [Secrets Store](core/secrets-store.md)
+
+## Actors
+
+An SDK MIGHT implement the Actors building block.
```

Comment:
If we create new inter-dependencies I'd love for some ceremonious declaration of this.

Reply:
Do you mean compatibilities between Actor invocations cross languages?

Reply:
Why is this MIGHT and not MUST?  What does not `MIGHT` mean? It's not part of the initial notational convention.

Reply:
The notation is defined above. I put MIGHT because Actors is kind of a separate thing from the "core APIs".

Reply:
I am changing it to MUST until we decide to make it optional. It seems to be the less controversial way to go now.